saleor/product/utils.py

```diff
@@ -151,3 +155,15 @@ def price_as_dict(price):
 def price_range_as_dict(price_range):
     return {'maxPrice': price_as_dict(price_range.max_price),
             'minPrice': price_as_dict(price_range.min_price)}
+
+
+def get_variant_url_from_product(product, attributes):
+    url = product.get_absolute_url() + '?'
+    for key, value in attributes.items():
+        url += '%s=%s' % (quote(key), quote(value))
+
+    return url
+
+
+def get_variant_url(variant):
+    return get_variant_url_from_product(variant.product, variant.attributes)
```

Comment:
We should kill all instances of `get_absolute_url`. It's just wrong for models to decide their URLs.